src/internal/kopia/upload_test.go

```diff
@@ -3017,3 +3025,783 @@ func (suite *HierarchyBuilderUnitSuite) TestBuildDirectoryTreeSelectsMigrateSubt
+	//           - file8
+	getBaseSnapshot := func() (fs.Entry, map[string]*int) {
+		counters := map[string]*int{}
+
+		folder, count := newMockStaticDirectory(
+			encodeElements(folderID3)[0],
+			[]fs.Entry{
+				virtualfs.StreamingFileWithModTimeFromReader(
+					encodeElements(fileName5)[0],
+					time.Time{},
+					newBackupStreamReader(
+						serializationVersion,
+						io.NopCloser(bytes.NewReader(fileData5))),
+				),
+				virtualfs.StreamingFileWithModTimeFromReader(
+					encodeElements(fileName6)[0],
+					time.Time{},
+					newBackupStreamReader(
+						serializationVersion,
+						io.NopCloser(bytes.NewReader(fileData6))),
+				),
+			})
+		counters[folderID3] = count
+
+		folder, count = newMockStaticDirectory(
+			encodeElements(folderID2)[0],
+			[]fs.Entry{
+				virtualfs.StreamingFileWithModTimeFromReader(
+					encodeElements(fileName3)[0],
+					time.Time{},
+					newBackupStreamReader(
+						serializationVersion,
+						io.NopCloser(bytes.NewReader(fileData3))),
+				),
+				virtualfs.StreamingFileWithModTimeFromReader(
+					encodeElements(fileName4)[0],
+					time.Time{},
+					newBackupStreamReader(
+						serializationVersion,
+						io.NopCloser(bytes.NewReader(fileData4))),
+				),
+				folder,
+			})
+		counters[folderID2] = count
+
+		folder, count = newMockStaticDirectory(
+			encodeElements(folderID1)[0],
+			[]fs.Entry{
+				virtualfs.StreamingFileWithModTimeFromReader(
+					encodeElements(fileName1)[0],
+					time.Time{},
+					newBackupStreamReader(
+						serializationVersion,
+						io.NopCloser(bytes.NewReader(fileData1))),
+				),
+				virtualfs.StreamingFileWithModTimeFromReader(
+					encodeElements(fileName2)[0],
+					time.Time{},
+					newBackupStreamReader(
+						serializationVersion,
+						io.NopCloser(bytes.NewReader(fileData2))),
+				),
+				folder,
+			})
+		counters[folderID1] = count
+
+		folder2, count := newMockStaticDirectory(
+			encodeElements(folderID4)[0],
+			[]fs.Entry{
+				virtualfs.StreamingFileWithModTimeFromReader(
+					encodeElements(fileName7)[0],
+					time.Time{},
+					newBackupStreamReader(
+						serializationVersion,
+						io.NopCloser(bytes.NewReader(fileData7))),
+				),
+				virtualfs.StreamingFileWithModTimeFromReader(
+					encodeElements(fileName8)[0],
+					time.Time{},
+					newBackupStreamReader(
+						serializationVersion,
+						io.NopCloser(bytes.NewReader(fileData8))),
+				),
+			})
+		counters[folderID4] = count
+
+		return baseWithChildren(
+				prefixFolders,
+				[]fs.Entry{
+					folder,
+					folder2,
+				},
+			),
+			counters
+	}
+
+	table := []struct {
+		name                  string
+		inputCollections      func(t *testing.T) []data.BackupCollection
+		inputExcludes         *pmMock.PrefixMap
+		expected              *expectedNode
+		expectedIterateCounts map[string]int
+	}{
+		{
+			// Test that even if files are excluded in the subtree selective subtree
+			// pruning skips traversing the it during hierarchy merging and the file
```

Comment:
nit: Could you improve this comment a bit?

Reply:
sure thing!